### DISCUSSION TITLE
QuerryParams adds to URL params with "None" value

### DISCUSSION THREAD

**Discussion by stevstrong** [encode/httpx#2194] [Potential Issue]
Hi there,

I used the `build_request` method of `AsyncClient()` with the following `params`:
`{'start_time': 1651222800, 'end_time': None}`

The resulting URL looks like this:
`<my_url>?start_time=1651222800&end_time=`

I would expect that the second key having value "None" from the `params` will not be added at all to the URL.

I tested the `prepare()` method of the standard requests lib and that returns the following URL (conforming with my expectations not adding the second key as it has "None" value):
`<my_url>?start_time=1651222800`

Digging in the source files I reached these lines in `__init__` method of `QueryParams` class:
https://github.com/encode/httpx/blob/3350d7e6831e2d942f68653b2f58cfa5ecb0bacd/httpx/_urls.py#L563-L593

Is it possible that a "None" value is not checked and excluded at all?
If yes, is that wanted?

**Comment by lovelydinosaur** [comment]
> I would expect that the second key having value "None" from the params will not be added at all to the URL.

We have different behaviour from `requests` here, yes.

Personally the "omit a parameter if the value is None" seems like non-intuitive behaviour to me, although I realise that folks *may* have different expectations here.

There's a brief note here: https://www.python-httpx.org/compatibility/#query-parameters

**Reply by lovelydinosaur** [reply]
I do think we could improve the documentation phrasing here to make the behaviour more clear...

> The `requests` package omits params where the values are `None`. For example `requests.get("https://www.example.com/", params={"search": None})` will send a request to the URL `https://www.example.com/`.
>
> In contrast to this, the `httpx` package includes an empty value for params where the values are None.  For example `httpx.get("https://www.example.com/", params={"search": None})` will send a request to the URL `https://www.example.com/?search=`.
>
> Overall we consider this to be more consistent behaviour.

**Reply by stevstrong** [reply]
I see, thanks. In this case It would definitely make sense to update the docs.

I can (and will) change my code to be inline with the new description, I just thought that `httpx` looks to keep compatibility with `requests` lib as far as possible.
If not, it is also fine for me.
